lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -340,3 +345,63 @@ def dump_grading_context(course):
     msg += "length=%d\n" % len(gcontext['all_descriptors'])
     msg = '<pre>%s</pre>' % msg.replace('<', '&lt;')
     return msg
+
+
+def iterate_problem_components(course):
+    """ Helper function for iterating through all problems of a course """
+    for section in course.get_children():
+        section_name = section.display_name_with_default
+        for subsection in section.get_children():
+            subsection_name = subsection.display_name_with_default
+            for unit in subsection.get_children():
+                unit_name = unit.display_name_with_default
+
+                parent_metadata = [section_name, subsection_name, unit_name]
+                for component in unit.get_children():
+                    if component.category == 'problem':
+                        yield component, parent_metadata
+
+
+def student_responses(course):
+    """
+    Yields student responses for all problems in course for writing out to a CSV file.
+    """
+    order = 1
+    for problem_component, parent_metadata in iterate_problem_components(course):
+        problem_component_info = (parent_metadata +
+                                  [problem_component.display_name_with_default, order, problem_component.location])
+        modules = StudentModule.objects.filter(
+            course_id=course.id,
+            grade__isnull=False,
+            module_state_key=problem_component.location,
+        ).order_by('student__username')
+        if modules:
+            has_answer = False
+            for module in modules:
+                try:
+                    state_dict = json.loads(module.state) if module.state else {}
+                    raw_answers = state_dict.get("student_answers", {})
+                except ValueError:
+                    log.error(
+                        "Student responses: Could not parse module state for " +
+                        "StudentModule id={module_id}, course={course_id}".format(
+                            module_id=module.id, course_id=course.id
+                        )
+                    )
+                    continue
+                pretty_answers = u', '.join(
+                    u"{problem}={answer}".format(problem=problem, answer=answer)
+                    for (problem, answer) in raw_answers.items()
+                )
+                yield problem_component_info + [module.student.username, pretty_answers]
+                if not has_answer:
+                    has_answer = True
+            if has_answer:
```

Comment:
Barring no errors up to this point, this problem has yielded at least one response. so we need to increment the order variable for the next problem.